### ISSUE TITLE
Terminal error with some apps on refocus on MacBooks with Touchbar

### ISSUE THREAD

**Issue by freakboy3742** [beeware/toga#308]
If you have a MacBook with a TouchBar, and you run Tutorial 0 (or any other app with no widget that can accept keyboard focus), then switch away from the app, and the then give the app focus again, you'll see the following error on the console:

```
2017-12-11 12:37:36.361 python[99397:90028276] IMKInputSession presentFunctionRowItemTextInputViewWithEndpoint:completionHandler: : [self textInputContext]=0x0 *NO* NSRemoteViewController to client, NSError=Error Domain=NSCocoaErrorDomain Code=4099 "The connection from pid 0 was invalidated from this process." UserInfo={NSDebugDescription=The connection from pid 0 was invalidated from this process.}, com.apple.inputmethod.EmojiFunctionRowItem
```

The error is caused because the Touchbar is looking for an input widget to use as the source for content tips - but no such input exists. 

I'm guessing we need to add some sort of "generic" event handler at a window or app level to acknowledge that the TouchBar exists and Toga isn't going to do anything with it (at least for now, anyway)

**Comment by freakboy3742** [comment]
If you *do* have a TextInput in your app, there's a related fix that is needed - see lines 6-12 of [this patch](https://github.com/pybee/toga/pull/300/files#diff-ab225d70a420183985f95985b1408c3a). This might give you some pointers to the fix for other widgets.

**Comment by amitsin6h** [comment]
I want to work on this, for my first contribution

**Comment by jmason86** [comment]
Not able to reproduce despite a variety of tests. Maybe a recent macOS patch has resolved this? 

Here's what I tried: 

1. Straight copy/paste of the [tutorial 0 instructions](http://toga.readthedocs.io/en/latest/tutorial/tutorial-0.html). Launched with ```python -m``` in the terminal, switched focus from/to the app several times, never saw the IMKInputSession error. 
2. Used ```beeware new``` to load up the default template. Then built in the same hello world functionality of tutorial 0. Added a ``` if __name__ == '__main__':``` to it so I could launch again with ```python -m```. Same result as before. 
3. Used ```beware build macos``` to create a .app. Launched the app from Finder and from Terminal with ```open```, watching for the error in Console. Didn't show up. Then also drilled down in Terminal through macOS/Hello World.app/Contents/MacOS/ and did a ```./Hello\ World``` so I could watch for the error in Terminal but still did not receive it. 

macOS 10.13.4
MacBook Pro (15-inch, Late 2016) with Touchbar
